### PR TITLE
Disable mount options when running --privileged

### PR DESCRIPTION
**Pull request by rhatdan**
We now default to setting storage options to "nodev", when running
privileged containers, we need to turn this off so the processes can
manipulate the image.

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
@cgwalters PTAL

File: libpod/runtime.go [hunk 1]
```diff
@@ -714,3 +715,15 @@ func (r *Runtime) generateName() (string, error) {
...
+// RemoveMountOptions removes mount ouptions from the graphdriver
+func (r *Runtime) RemoveMountOptions() {
+	var opts []string
+	for _, i := range r.config.StorageConfig.GraphDriverOptions {
+		if strings.Contains(i, "mountopt") {
+			continue
+		}
+		opts = append(opts, i)
+	}
+	r.config.StorageConfig.GraphDriverOptions = opts
```
Comment (mheon) [inline libpod/runtime.go]: Can we do this without modifying the storage.Store?

**Comment by mheon** [comment]
Would be a lot better if we had an API to toggle per-container in c/storage - this will make it so that launching a single container will disable mount opts for the entire container once we move CRI-O to libpod

**Comment by TomSweeneyRedHat** [comment]
bot, retest this please

File: libpod/runtime.go (hunk 1 above)
Comment (rhatdan) [reply libpod/runtime.go]: Not sure how?

**Comment by rhatdan** [comment]
Opened https://github.com/containers/storage/pull/211 so we could pass this down the the graphdriver per container,

File: libpod/runtime.go (hunk 1 above)
Comment (rhatdan) [reply libpod/runtime.go]: It no longer modifies the store.  Just sends the options in on the container creation.

File: libpod/runtime.go [hunk 2]
```diff
@@ -714,3 +715,15 @@ func (r *Runtime) generateName() (string, error) {
 func (r *Runtime) ImageRuntime() *image.Runtime {
 	return r.imageRuntime
 }
+
+// RemoveMountOptions removes mount ouptions from the graphdriver
```
[the remark this answers is not in the document]
Comment (rhatdan) [reply libpod/runtime.go]: Removed

**Comment by rhatdan** [comment]
Requires https://github.com/containers/storage/pull/226

**Comment by rhatdan** [comment]
@cgwalters @mheon @baude @umohnani8 @giuseppe @vrothberg PTAL
I believe this is ready to merge.

**Comment by mheon** [comment]
LGTM

**Comment by TomSweeneyRedHat** [comment]
LGTM

**Pull request merged**